### PR TITLE
Docker: Added warning if network could not be found

### PR DESCRIPTION
**Pull request by zweizeichen**
Closes #1308 - avoids rebase entanglements and general time-travel-related corruption :)

### CODE REVIEW

File: provider/docker.go [hunk 1]
```diff
@@ -412,6 +413,8 @@ func (provider *Docker) getIPAddress(container dockerData) string {
 			if network != nil {
 				return network.Addr
 			}
+
+			log.Warn("Could not find network named '%s' for container '%s'! Maybe you're missing the project's prefix in the label? Defaulting to first available network.", label, container.Name)
```
Comment (timoreimann) [inline provider/docker.go]: Needs to be `Warnf`.

**Comment by timoreimann** [comment]
Closes #1156.

File: provider/docker.go (hunk 1 above)
Comment (zweizeichen) [inline provider/docker.go]: Oops, I always mix those up

**Comment by zweizeichen** [comment]
We could try to detect if the instance of traefik runs under a prefix and then could try to find the missing network by prepending the prefix. If you both have a network `proxy` and `prefix_proxy`, but both instances try to connect on network `proxy` - is this intentional or just a configuration mistake like in #1156? That situation would be interesting to resolve. If we got two potentially matching networks IMHO we should go with the non-prefixed version and print a warning (it could be unintended) as otherwise you could not explicitly use the non-prefixed version at all.

**Comment by timoreimann** [comment]
@zweizeichen definitely sounds like a more sophisticated solution will take some time.

Agree we should get this PR merged now and worry about more cleverness in a separate issue. :+1:

**Comment by timoreimann** [comment]
@emilevauge PTAL. :-)

**Pull request merged**